libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -161,6 +165,20 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishDRepPulser (nes ^. newEpochStateGovStateL . drepPulsingStateGovStateL)
 
+-- | Query pool stake distribution.
+querySPOStakeDistr ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  Set (KeyHash 'StakePool (EraCrypto era)) ->
+  -- | Specify pool key hashes whose stake distribution should be returned. When this set is
+  -- empty, distributions for all of the pools will be returned.
+  (Map (KeyHash 'StakePool (EraCrypto era)) (IndividualPoolStake (EraCrypto era)))
+querySPOStakeDistr nes keys
+  | null keys = distr
+  | otherwise = distr `Map.restrictKeys` keys
+  where
+    distr = psPoolDistr . fst $ finishDRepPulser (nes ^. newEpochStateGovStateL . drepPulsingStateGovStateL)
```

Comment:
I'll probably factor out the `finishDRepPulser (nes ^. newEpochStateGovStateL . drepPulsingStateGovStateL)` since that is used elsewhere too.